packages/react/src/components/TreeView/TreeView.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, match, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function TreeView({
+  active: prespecifiedActive,
```

Comment:
Is the hope that consumers could use this in a controlled/uncontrolled fashion like in other components?

Reply:
yes, for controlled/uncontrolled active nodes (and similarly for selected nodes)